src/player/video_player.py

```diff
@@ -349,11 +352,14 @@ def data_source(self, data_source):
                     video_width[monitor] = int(dimension[0])
                     video_height[monitor] = int(dimension[1])
             except subprocess.CalledProcessError:
-                video_width[monitor] = None
-                video_height[monitor] = None
-            
+                for monitor, video in data_source.items():
+                    video_width.setdefault(monitor, None)
+                    video_height.setdefault(monitor, None)
+                    
             for (monitor, window) in self.windows.items():
-                media = window.media_new(data_source[monitor.get_model()])
+                source = data_source['Default'] if data_source[monitor.get_model()] == "" else  data_source[monitor.get_model()]
```

Comment:
string check for empty strings must be `len(data_source[monitor.get_model()]) == 0`